### DISCUSSION TITLE
How do I create a flow that allows users to sign in once they've verified their email, but also require a phone verification at a later stage?

### DISCUSSION THREAD

**Discussion by justinfarrelldev** [supabase/supabase#27889] [Questions] [answered]
Here is my desired flow: 
- User provides their email, phone number and password all at once in my sign-up component
- The user is prompted to verify their email (and they do so)
- Once they're redirected back to my application, they are prompted to verify their phone number
- Once they verify THAT, they are then allowed access to my application

I know all the Remix parts, I just need help with the '@supabase/ssr' parts of it!

The major hitch I've encountered is that when calling the following: 
```typescript
        const { error } = await supabaseClient.auth.signUp({
            email,
            password,
            phone,
            options,
        });
```

...it seems that the phone number is still null (both email and password are set correctly, though): 
![image](https://github.com/supabase/supabase/assets/42923203/8ca1bcf7-17c4-4605-9b6c-131dff9b17df)

I am absolutely sure that the "phone" variable is correctly set (confirmed via logs).

I assume once "phone" is actually present, it would be relatively straightforward to verify the phone number manually as well? That way, when the user is redirected from the email, they can click a button or something to do SMS verification.

Also, to answer the question that will definitely be asked (reasonably), I would like both email and SMS verification since the application I am creating would be relatively vulnerable to botting if someone took the time to create many accounts (and I frankly don't want to even chance it).

Thank you!

**Comment by j4w8n** [comment, marked as answer]
You can only use one or the other to signup a user with this method. The auth-js code checks for the existence of an `email` first, then does an `else if` statement for `phone`.

You should be able to sign them up with an email and password, then after verification and login on your app, have them put in their phone number. Then, with the phone number, just use `supabase.auth.updateUser({ phone })` and they should be sent an OTP to verify it. After that, I assume they could login with either email or phone, but I've never tried it.

Or if the phone is what's really helping you to prevent bots, then do the reverse.

**Reply by justinfarrelldev** [reply]
Awesome, thank you for the detailed answer. I will do that!
